### PR TITLE
fix(localize): get fraction digits from the current locale currencies instead of the default English

### PR DESCRIPTION
**Pull request by mostafabita**
When a currency is customized in a locale (ex. fa) and the number of fraction digits is changed, Angular ignores it when using the currency pipe and uses its default in English. This problem becomes more apparent when we add a custom currency that does not have fraction digits and Angular uses DEFAULT_NB_OF_CURRENCY_DIGITS instead of the defined number

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?
When a currency is customized in a locale (ex. fa) and the number of fraction digits is changed, Angular ignores it when using the currency pipe and uses its default in English. This problem becomes more apparent when we add a custom currency that does not have fraction digits and Angular uses DEFAULT_NB_OF_CURRENCY_DIGITS instead of the defined number

Issue Number: N/A

## What is the new behavior?
Using overridden fraction digits instead of the default in English or DEFAULT_NB_OF_CURRENCY_DIGITS when displaying a localized currency

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by dylhunn** [comment]
@mostafabita It looks like you have some failing tests you'll need to get green.

This sounds reasonable in principle. It may be breaking, so we'll need to run a TGP, and keep a close eye on it after it lands (a lot of i18n support is not well unit-tested).

File: packages/common/test/i18n/locale_data_api_spec.ts [hunk 1]
```diff
@@ -52,6 +52,7 @@ import {FormatWidth, FormStyle, getCurrencySymbol, getLocaleDateFormat, getLocal
         expect(getNumberOfCurrencyDigits('USD')).toEqual(2);
         expect(getNumberOfCurrencyDigits('GNF')).toEqual(0);
         expect(getNumberOfCurrencyDigits('BHD')).toEqual(3);
+        expect(getNumberOfCurrencyDigits('USD', 'fr')).toEqual(2);
```
Comment (AndrewKushnir) [inline packages/common/test/i18n/locale_data_api_spec.ts:55]: @mostafabita thanks for the fix!

Could you please also add a test for the `CurrencyPipe` class in packages/common/test/pipes/number_pipe_spec.ts that would illustrate the original problem and the fix?

Thank you.

Comment (mostafabita) [reply packages/common/test/i18n/locale_data_api_spec.ts:55]: @AndrewKushnir Yes, sure!

**Comment by AndrewKushnir** [comment]
@mostafabita thanks for additional changes. It looks like CI is failing for 2 reasons:

- one of the commits has a commit message that is not aligned with the guidelines (see `lint` job failure)
- the public API symbol file is not up-to-date with the latest changes, so you'd need to run `yarn bazel run //packages/common:common_api.accept` as mentioned in the `test` CI job output

Please also rebase the PR on top of the latest main branch: there was an unrelated issue that is also causing some tests to fail in your branch and rebasing will help resolve that.

Thank you.

**Comment by jessicajaniuk** [comment]
Closing this since there's been no updates since September. Feel free to open a new PR if you'd like to see these changes land.

**Pull request closed without merging**